### PR TITLE
[AppConfig] AAD test enabled

### PR DESCRIPTION
**Pull request by mssfang**

### CODE REVIEW

File: sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/AadCredentialTest.java [hunk 1]
```diff
@@ -53,20 +54,37 @@ private void setup(HttpClient httpClient, ConfigurationServiceVersion serviceVer
...
                 .endpoint(endpoint)
-                .serviceVersion(serviceVersion);
-
+                .serviceVersion(serviceVersion)
+                .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS));
+            builder = setHttpClient(httpClient, builder);
             if (interceptorManager.isRecordMode()) {
-                builder.addPolicy(interceptorManager.getRecordPolicy()); // Record
+                builder.addPolicy(interceptorManager.getRecordPolicy())
+                    .addPolicy(new RetryPolicy()); // Record
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/AadCredentialTest.java:62]: The builder should already have a retry policy, won't this add a second one?

File: sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/AadCredentialTest.java [hunk 2]
```diff
@@ -53,20 +54,37 @@ private void setup(HttpClient httpClient, ConfigurationServiceVersion serviceVer
                 .httpClient(httpClient)
                 .credential(tokenCredential)
                 .endpoint(endpoint)
-                .serviceVersion(serviceVersion);
-
+                .serviceVersion(serviceVersion)
+                .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS));
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/AadCredentialTest.java:58]: I don't think we need to include this anymore. This doesn't configure any allowed headers and query parameters anyways.

File: sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/AadCredentialTest.java [hunk 3]
```diff
@@ -53,20 +54,37 @@ private void setup(HttpClient httpClient, ConfigurationServiceVersion serviceVer
...
             }
             client = builder.buildClient();
         }
     }
 
-    @Disabled("Disable it until resource is in production. The resource for testing is still a dogfood environment.")
+    ConfigurationClientBuilder setHttpClient(HttpClient httpClient, ConfigurationClientBuilder builder) {
+        if (interceptorManager.isRecordMode()) {
+            return builder
+                .httpClient(buildSyncAssertingClient(httpClient));
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/AadCredentialTest.java:71]: Don't we want to add the asserting client when running LIVE test mode as well?

Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/AadCredentialTest.java]: Yea. It should also apply to LIVE mode as well. Updated.

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**